Indexing error with unhelpful error message when a generated class is registered as an additional bean instead of a generated bean

**Describe the bug**
An indexing error with an unhelpful error message is displayed if you add a generated class
as an additional bean at build message

**Expected behavior**
An error message that suggests to use GeneratedBeanGizmoAdaptor .
Ex: "Unable to find class file for ${className}. Maybe ${className} was registered as an additional bean via `AdditionalBeanBuildItem`. In that case, remove the producer of `AdditionalBeanBuildItem` and use a GeneratedBeanGizmoAdaptor as the class output for ClassCreator."

**Actual behavior**
An unhelpful "java.lang.IllegalArgumentException: stream cannot be null" is shown
```
java.lang.IllegalArgumentException: stream cannot be null
[ERROR] 	at org.jboss.jandex.Indexer.index(Indexer.java:1584)
[ERROR] 	at io.quarkus.deployment.index.IndexingUtil.indexClass(IndexingUtil.java:32)
[ERROR] 	at io.quarkus.arc.deployment.BeanArchiveProcessor.build(BeanArchiveProcessor.java:73)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
[ERROR] 	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
[ERROR] 	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$2.execute(ExtensionLoader.java:936)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:277)
[ERROR] 	at org.jboss.threads.ContextClassLoaderSavingRunnable.run(ContextClassLoaderSavingRunnable.java:35)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor.safeRun(EnhancedQueueExecutor.java:2046)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.doRunTask(EnhancedQueueExecutor.java:1578)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1452)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:834)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:479)
```


**To Reproduce**
https://github.com/Christopher-Chianelli/issue-reproducer/tree/gizmo-cannot-be-indexed-error
`mvn clean install`; the error will occur during the build of the example project.

**Configuration**
```properties
# Add your application.properties here, if applicable.

```

**Screenshots**
(If applicable, add screenshots to help explain your problem.)

**Environment (please complete the following information):**
 - Output of `uname -a` or `ver`: 
 - Output of `java -version`: 
```
openjdk 11.0.9 2020-10-20
OpenJDK Runtime Environment 18.9 (build 11.0.9+11)
OpenJDK 64-Bit Server VM 18.9 (build 11.0.9+11, mixed mode, sharing)
```
 - GraalVM version (if different from Java): 
 - Quarkus version or git rev: 1.9.2.Final
 - Build tool (ie. output of `mvnw --version` or `gradlew --version`): 
```
Apache Maven 3.6.3 (cecedd343002696d0abb50b32b541b8a6ba2883f)
```

**Additional context**
(Add any other context about the problem here.)

--- a/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/AdditionalBeanBuildItem.java
+++ b/extensions/arc/deployment/src/main/java/io/quarkus/arc/deployment/AdditionalBeanBuildItem.java
@@ -21,6 +21,11 @@ import io.quarkus.builder.item.MultiBuildItem;
  * An additional bean may have the scope defaulted via {@link #defaultScope} and {@link Builder#setDefaultScope(DotName)}. The
  * default scope is only used if there is no scope declared on the bean class. The default scope should be used in cases where a
  * bean class source is not controlled by the extension and the scope annotation cannot be declared directly on the class.
+ *
+ * <h2>Generated Classes</h2>
+ *
+ * This build item should never be produced for a generated class - {@link GeneratedBeanBuildItem} and
+ * {@link GeneratedBeanGizmoAdaptor} should be used instead.
  */
 public final class AdditionalBeanBuildItem extends MultiBuildItem {
 

